lib/web_ui/lib/src/engine/raw_keyboard.dart

```diff
@@ -96,6 +96,14 @@ class RawKeyboard {
     final FlutterHtmlKeyboardEvent event = FlutterHtmlKeyboardEvent(domEvent as DomKeyboardEvent);
     final String timerKey = event.code!;
 
+    // During IME composition, Tab fires twice (once for composition and once
+    // for regular tabbing behavior), which causes issues. Intercepting the
+    // tab keydown event during composition prevents these issues from occurring.
+    // https://developer.mozilla.org/en-US/docs/Web/API/Element/keydown_event#ignoring_keydown_during_ime_composition
+    if (event.type == 'keydown' && event.key == 'Tab' && event.isComposing) {
+      return;
+    }
```

Comment:
I suggest putting this logic in a function:
```suggestion
    if (_shouldIgnore(event)) {
      return;
    }
```
Then add:
```dart
bool _shouldIgnore(FlutterHtmlKeyboardEvent event) {
  // During IME composition, Tab fires twice (once for composition and once
  // for regular tabbing behavior), which causes issues. Intercepting the
  // tab keydown event during composition prevents these issues from occurring.
  // https://developer.mozilla.org/en-US/docs/Web/API/Element/keydown_event#ignoring_keydown_during_ime_composition
  return event.type == 'keydown' && event.key == 'Tab' && event.isComposing;
}
```